src/commonTest/resources/dev/rhovas/interpreter/programs/rosettacode/Fibonacci.rho

```diff
@@ -0,0 +1,26 @@
+func pfib(n: Integer): Integer {
+    require n >= 0;
+
+    var lastFib = -1;
+    var fib = 1;
+    range(0, n, :incl).for {
+        var sum = fib + lastFib;
+        lastFib = fib;
+        fib = sum;
+    };
+    return fib;
+}
+
+func fib(n: Integer): Integer {
+    match ([n < 0, n.mod(2)]){
+        [true, 0]: return -pfib(-n);
+        [true, _]: return  pfib(-n);
+        else:      return  pfib( n);
```

Comment:
Don't vertical align. It may look nicer at a glance, but it's a lot less maintainable and causes cascading diffs from the reformatting. My hope is that the use cases for this can largely be addressed by editor customization.

Reply:
Dang. FizzBuzz has about the most beautiful vertical alignment. For a bit there I thought that vertical alignment was one of the features you were pushing for, with the way that match can beautify complex logic.

`
range(1, 15, :incl).for {
    match ([val.mod(3), val.mod(5)]) {
        [0, 0]: print("FizzBuzz");
        [0, _]: print("Fizz");
        [_, 0]: print("Buzz");
        [_, _]: print(val);
    }
}
`

And yes this is definitely the correct version of FizzBuzz with zero modifications whatsoever.

Reply:
Recall the original version has `else:`, which is one character less than the other cases:

```scala
range(1, 15, :incl).for {
    match ([val.mod(3), val.mod(5)]) {
        [0, 0]: print("FizzBuzz");
        [0, _]: print("Fizz");
        [_, 0]: print("Buzz");
        else: print(val);
    }
}
```

My thoughts here are largely shared with [Google's Java Style Guide](https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment). Specifically, though it may improve readability it also causes problems for maintainability as changes can require re-aligning unrelated parts of the code.

Long-term, my vision is that a source file is just an AST representation and the client editor can display that however the user desired. That includes simple styling like whitespaces/indentation, but eventually more complex things like rendering doc comments, DSLs (e.g. mathematics), and so on. [JetBrains MPS](https://www.jetbrains.com/mps/) is an interesting project in this regard using a projectional editor.